### DISCUSSION TITLE
sharing queryclient between packages

### DISCUSSION THREAD

**Discussion by Itaysade98** [TanStack/query#3976] [Q&A]
lets say i'm working on react project A that utilizes components and hooks from package B that i developed. these components and hooks request data from an API, and i wanted to use useQuery to handle these requests - so i wrapped the main component in project A with a QueryClientProvider but i get an error that there is no queryclient set even though the components and hooks that use useQuery are called inside the provider. i've looked for similar problems and had no success, tried to configure react-query as a peer-dependency in package B, made sure it wasn't version differences between A and B and enabled contextSharing in the provider. if anyone has a suggestion i'd love to try.

**Comment by kxxoling** [comment]
Have you tried monorepo? eg: https://pnpm.io/workspaces
